lightningd/pay.c

```diff
@@ -1496,6 +1521,77 @@ static const struct json_command listsendpays_command = {
 };
 AUTODATA(json_command, &listsendpays_command);
 
+
+static struct command_result *json_delpay(struct command *cmd,
+						const char *buffer,
+						const jsmntok_t *obj UNNEEDED,
+						const jsmntok_t *params)
+{
+	struct json_stream *response;
+	const struct wallet_payment **payments;
+	const char *status_str;
+	enum wallet_payment_status status;
+	struct sha256 *payment_hash;
+
+	if (!param(cmd, buffer, params,
+		p_req("payment_hash", param_sha256, &payment_hash),
+		p_opt("status", param_string, &status_str),
+		NULL))
+		return command_param_failed();
+
+	if (!payment_hash)
+		return command_fail(cmd, JSONRPC2_INVALID_PARAMS,
+						"missing required parameter: payment_hash");
+
+
+	status = status_str == NULL ? PAYMENT_COMPLETE : string_to_payment_status(status_str);
+
+	if (status == -1)
+		return command_fail(cmd, JSONRPC2_INVALID_PARAMS,
+						"status insert %s wrong", status_str);
+
+	if (status == PAYMENT_PENDING)
+		return command_fail(cmd, PAY_STATUS_UNEXPECTED, "invalid status: %s",
```

Comment:
This is an error in the argument, not in the status of the actual payment, so should be `JSONRPC2_INVALID_PARAMS`.

Reply:
Changed in https://github.com/ElementsProject/lightning/commit/627a374820dd03876eae9e4a99d661cb39f63934